tasks/main.yml

```diff
@@ -0,0 +1,39 @@
+---
+
+- name: Find required providers
+  set_fact:
+    __certificate_providers: |
+      {{
+        certificate_requests |
+        map(attribute='provider', default=__certificate_provider_default) |
+        unique |
+        list
+      }}
+
+- name: Ensure provider packages are installed
+  package:
+    name: "{{ __certificate_provider_vars[__certificate_provider].packages }}"
```

Comment:
I think you need to add the `__certificate_packages` to this - something like
```
     name: "{{ __certificate_provider_vars[__certificate_provider].packages + __certificate_packages }}"
```

Reply:
That's right! :facepalm: 
I will do it in a separe play since this is installing provider specific packages. Otherwise the certificate packages will be installed multiple times (actually the attempt to install will happen multiple times).

Reply:
What do you mean by "separate play"?

Reply:
> What do you mean by "separate play"?

separate tasks:
```yaml

- name: Ensure certificate role dependencies are installed
  package:
    name: "{{ __certificate_packages }}"
    state: present
...

- name: Ensure provider packages are installed
  package:
    name: "{{ __certificate_provider_vars[__certificate_provider].packages }}"
    state: present
  loop: "{{ __certificate_providers }}"
  loop_control:
    loop_var: __certificate_provider
  when: |
    __certificate_provider_vars[__certificate_provider].packages is defined
```